### DISCUSSION TITLE
Replace nearly transparent pixels with specific color.

### DISCUSSION THREAD

**Discussion by neilliang** [ImageMagick/ImageMagick#4301] [General]
Hey,
I would like to know that how to replace nearly transparent pixels with specific color.
For example, The png image below contains some invisible pixels that can't be seen thru human's eyes.
![123](https://user-images.githubusercontent.com/1026331/135595795-ec2e36bc-9f86-4b93-b709-0b8a9b7070d7.png)

I added black background so we can recognize it easier.
![input](https://user-images.githubusercontent.com/1026331/135598299-e47e37c8-06c3-41ca-bd1d-5b528f78dac1.JPG)

After I replace all the colors with white with command:
**_magick input.png -fill "#FFFFFF" +opaque none  -alpha off result.png_**

The result was shown below:
![_123](https://user-images.githubusercontent.com/1026331/135597804-a10ebf30-81f8-4c2f-8b18-e8405e1ad1d9.png)

How to replace it with specific color for only the pixels contains transparent attribute? Thank you.

**Comment by snibgo** [comment]
> I would like to know that how to replace nearly transparent pixels with specific color.

What is your definition of "nearly transparent"?

You could make a mask using whatever definition you want. For example, extract the alpha and make a mask that is white where the alpha is in a defined range, otherwise black. Then use that mask to change pixels to your required colour.

**Comment by fmw42** [comment]
Those pixels are not transparent. They are just darker than you can see. So you can just threshold your image and composite the original with some solid color image using the mask to control where the color is placed.

**Comment by fmw42** [comment]
Perhaps this is what you want.

`convert animal.png -background red -alpha background -alpha off animal_red.png`

![animal_red](https://user-images.githubusercontent.com/13458218/136257000-6cff72e1-27b5-40d4-99d4-004bcc3f6c0a.png)

If on ImageMagick 7, replace convert with magick.
